packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart

```diff
@@ -29,81 +29,76 @@ const String timelineScreenId = 'timeline';
 class TimelineController implements DisposableController {
   TimelineController() {
     timelineService = TimelineService(this);
-    fullTimeline = FullTimeline(this);
-    frameBasedTimeline = FrameBasedTimeline(this);
-    timelines = [frameBasedTimeline, fullTimeline];
+    processor = TimelineProcessor(this);
   }
 
   final cpuProfilerController = CpuProfilerController();
 
   /// Notifies that a timeline event was selected.
-  ValueListenable get selectedTimelineEventNotifier =>
-      _selectedTimelineEventNotifier;
+  ValueListenable get selectedTimelineEvent => _selectedTimelineEventNotifier;
   final _selectedTimelineEventNotifier = ValueNotifier<TimelineEvent>(null);
 
+  /// Notifies that a timeline frame has been selected.
+  ValueListenable get selectedFrame => _selectedFrameNotifier;
+  final _selectedFrameNotifier = ValueNotifier<TimelineFrame>(null);
+
+  /// Notifies when an empty timeline recording finishes
+  ValueListenable get emptyRecording => _emptyRecordingNotifier;
+  final _emptyRecordingNotifier = ValueNotifier<bool>(false);
+
+  /// Notifies that the timeline is currently being recorded.
+  ValueListenable get recording => _recordingNotifier;
```

Comment:
@pq refactoring microfriction: it would be nice that when you renamed a property, the private field associated with it also renamed.
In this case the property was original `recordingNotifier => _recordingNotifier`
but refactoring the name of `recordingNotifier` didn't rename `_recordingNotifier`